pkg/webhooks/hiveextension/v1beta1/agentclusterinstall_validation_hook.go

```diff
@@ -314,12 +314,16 @@ func areImageSetRefsEqual(imageSetRef1 *hivev1.ClusterImageSetReference, imageSe
 }
 
 func isUserManagedNetworkingSetToFalseWithSNO(newObject *hiveext.AgentClusterInstall) bool {
-	return newObject.Spec.ProvisionRequirements.ControlPlaneAgents == 1 &&
-		newObject.Spec.ProvisionRequirements.WorkerAgents == 0 &&
+	return isSNO(newObject) &&
 		newObject.Spec.Networking.UserManagedNetworking != nil &&
 		!*newObject.Spec.Networking.UserManagedNetworking
 }
 
+func isSNO(newObject *hiveext.AgentClusterInstall) bool {
+	return newObject.Spec.ProvisionRequirements.ControlPlaneAgents == 1 &&
+		newObject.Spec.ProvisionRequirements.WorkerAgents == 0
```

Comment:
`newObject.Spec.ProvisionRequirements.ControlPlaneAgents == 1` is enough to determine SNO, as SNO is (starting 4.11) used to refer to single control-plane node clusters.

In the future we may want to have day-1 SNO+workers installation (currently it's only day-2 for adding workers to SNO), so better have this condition be prepared for that